### PR TITLE
[PREVIEW] Feature/sl 1001/s2 s authentication

### PR DESCRIPTION
**Pull request by leszekgu**
Waiting for events and API approval before merge

### CODE REVIEW

File: src/test/integration/uk/gov/hmcts/reform/sandl/snlrules/testcategories/IntegrationTestWithFakeRules.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package uk.gov.hmcts.reform.sandl.snlrules.testcategories;
+
+public interface IntegrationTestWithFakeRules {
```
Comment (r-kruk-kainos) [inline src/test/integration/uk/gov/hmcts/reform/sandl/snlrules/testcategories/IntegrationTestWithFakeRules.java]: remove

File: src/test/java/uk/gov/hmcts/reform/sandl/snlrules/JwtTokenHelper.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.http.HttpHeaders;
+import uk.gov.hmcts.reform.sandl.snlrules.security.S2SAuthenticationService;
+
+import java.util.Date;
+
+public class JwtTokenHelper {
+
```
Comment (r-kruk-kainos) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlrules/JwtTokenHelper.java]: This class should probably be removed as I don't see any usages.

Comment (leszekgu) [reply src/test/java/uk/gov/hmcts/reform/sandl/snlrules/JwtTokenHelper.java]: it is by integration tests, moved there.

File: src/test/resources/application.yaml [hunk 3]
```diff
@@ -7,6 +7,9 @@ drools:
 management:
   security:
     enabled: ${MANAGEMENT_SECURITY_ENABLED:true}
+    rules:
+      jwtSecret: "FakeTestSecret"
```
Comment (radoslawlandowskikainos) [inline src/test/resources/application.yaml]: Make sure that the secrets you use will work with real services. It is important for integration tests which will fail to authenticate if secrets are incorrect

**Pull request merged**